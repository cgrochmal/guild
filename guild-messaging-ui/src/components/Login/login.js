import React from 'react'
import PropTypes from 'prop-types'
import {withRouter} from 'react-router-dom'

import UserList from '../UserList/userList'
import './login.scss'

class Login extends React.Component {
  constructor(props) {
    super(props)
    Login.propTypes = {
      users: PropTypes.arrayOf(PropTypes.object)
    }
    this.state = {
      selectedUser: null
    }

    // I prefer to use explicit binding to auto-bound arrow functions in my react components.
    // I'm happy to elaborate on reasons why in the follow-up interview
    this.handleUserSelected = this.handleUserSelected.bind(this)
    this.logIn = this.logIn.bind(this)
  }

  /**
   * EVENT HANDLERS
   */
  handleUserSelected(user) {
    this.setState({selectedUser: user})
  }
  logIn() {
    const {selectedUser} = this.state
    localStorage.currentUserId = selectedUser.id
    this.props.history.push('/select-chat')
  }

  /**
   * RENDER FUNCTIONS
   */
  render() {
    const {users} = this.props
    const {selectedUser} = this.state

    return (
      <section className='login-container'>
        <h1 className='login-container__heading'>Please select your username</h1>
        <UserList users={users} handleUserSelected={this.handleUserSelected}/>
        <button className='chat-button login-container__button' onClick={this.logIn} disabled={!selectedUser}>
          {'Log In'}
        </button>
      </section>
    )
  }
}

export default withRouter(Login)