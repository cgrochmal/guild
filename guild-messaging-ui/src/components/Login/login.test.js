import React from 'react'
import renderer from 'react-test-renderer'
import {Login} from './login.js'
import TestHelper from '../../test/testHelper'

it('matches the Login snapshot', () => {
  localStorage.currentUserId = "1"
  const mockUsers = TestHelper.getMockUsers(5, Date.now())
  const tree = renderer
    .create(<Login users={mockUsers}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})