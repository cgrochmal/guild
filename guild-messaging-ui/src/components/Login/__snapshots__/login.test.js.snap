// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Login snapshot 1`] = `
<section
  className="login-container"
>
  <h1
    className="login-container__heading"
  >
    Please select your username
  </h1>
  <div
    className="user-list"
  >
    <span
      className="user-list__row"
      onClick={[Function]}
    >
      user0
    </span>
    <span
      className="user-list__row"
      onClick={[Function]}
    >
      user1
    </span>
    <span
      className="user-list__row"
      onClick={[Function]}
    >
      user2
    </span>
    <span
      className="user-list__row"
      onClick={[Function]}
    >
      user3
    </span>
    <span
      className="user-list__row"
      onClick={[Function]}
    >
      user4
    </span>
  </div>
  <button
    className="chat-button login-container__button"
    disabled={true}
    onClick={[Function]}
  >
    Log In
  </button>
</section>
`;
